### ISSUE TITLE
[Question] For what iteration is the evaluation metric returned when using early stopping in Sklearn's GridSearchCV or RandomizedSearchCV

### ISSUE THREAD

**Issue by e-baumer** [microsoft/LightGBM#3595]
Thank you for developing and continuing to support a great tool.

This question is related to using early stopping when hyper-parameter tuning a lightGBM model along with Sklearn's GridSearchCV and RandomizedSearchCV.  GridSearchCV and RandomizedSearchCV rank the parameters based on a scoring metric. If one uses early stopping (as shown in the example below), is the model from the last iteration or the best iteration used to calculate the scoring metric in Sklearn's GridSearchCV or RandomizedSearchCV? If it's from the last iteration, is there a way to specify using the model from the best iteration?

```
x_train, x_val, y_train, y_val = train_test_split(
    x, y,
    test_size=0.2,
    random_state=42,
    stratify=y
)

kf = StratifiedKFold(n_splits=5, shuffle=True, random_state=42)

clf = lgb.LGBMClassifier(
    max_depth=-1,
    objective='cross_entropy',
    random_state=42,
    n_jobs=8,
    n_estimators=10000
)

fit_params = {
    "early_stopping_rounds": 100,
    "categorical_feature": np.arange(13, 18).tolist()
}

grid_params = {
    'learning_rate': np.round(np.linspace(0.010, 0.15, 20), 4),
    'feature_fraction': np.round(np.linspace(0.4, 0.99, 40), 4),
    'drop_rate': np.round(np.linspace(0.10, 0.80, 30), 4),

}

grid_search = RandomizedSearchCV(
    clf,
    grid_params,
    n_iter=5000,
    cv=kf,
    random_state=42
)

grid_search.fit(x_train, y_train, eval_set=[(x_val, y_val)], **fit_params)
```

Thank you for your consideration.

**Comment by btrotta** [comment]
It will choose the best iteration. Sklearn calculates the accuracy score using the estimator's `predict` method, which uses the best iteration (https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMModel.html#lightgbm.LGBMModel.predict)
